includes/rest-api/Controllers/Version3/class-wc-rest-controller.php

```diff
@@ -93,16 +93,53 @@ public function get_endpoint_args_for_item_schema( $method = WP_REST_Server::CRE
 			return $endpoint_args;
 		}
 
+		$endpoint_args = $this->adjust_wp_5_5_datatype_compatibility( $endpoint_args );
+
+		return $endpoint_args;
+	}
+
+	/**
+	 * Change datatypes `date-time` to string, and `mixed` to composite of all built in types.
+	 *
```

Comment:
Worth mentioning why we're doing this dynamically (the conflict between WP 5.4 and WP 5.5)?

Reply:
Added further explanations in comments in https://github.com/woocommerce/woocommerce/pull/27318/commits/3dd508fd356e44c8799ea7b29227ba04e6df2729